tar_scm.py

```diff
@@ -72,6 +72,17 @@ def fetch_upstream_git(url, clone_dir, revision, cwd, kwargs):
     """Fetch sources via git."""
     safe_run(['git', 'clone', url, clone_dir], cwd=cwd,
              interactive=sys.stdout.isatty())
+    if revision:
+        # check if the reference already exists.
+        try:
+            safe_run(['git', 'rev-parse', '--verify', '--quiet', revision],
```

Comment:
The revision check in `switch_revision_git` is slightly different because it checks against the `revs = [x + revision for x in ['origin/', '']]`. Since `fetch_upstream_git` does not call `switch_revision_git` it would be impossible to change the revision to 'origin/' + revision.